terminal/terminal.md

```diff
@@ -0,0 +1,417 @@
+
+`apprentice` is the current directory
+
+`danielvilla` is the name of the user
+
+`$` means you are a normal user
+
+the prompt is highly customizable changing the env variable **$PS1** but for now let's keep it the way it is
+
+Let's type whatever we want just to see what happen, let's experiment:
+```
+Daniels-MacBook-Air:apprentice danielvilla$ hey
+-bash: hey: command not found
+```
+As you can see the CLI *(command line interface)* is trying to resolve that name, it thinks it is a command, and as it wasn't found it return `command not found`, that tell you how to use it, and what it is expecting
+
+Now let's input a real command.
+
+Type `ls` to list all the files on the current directory. 
+
+```
+Daniels-MacBook-Air:apprentice danielvilla$ ls
+LICENSE   README.md
+```
+
+but wait how do we know in what directory we are? we use the command `pwd` that means *path of working directory*
+```
+Daniels-MacBook-Air:apprentice danielvilla$ pwd
+/Users/danielvilla/apprentice
+```
+Thre are two types of path, `absolute path` and `relative path` what pwd returns is the absolute path because it is NOT relative to the current directory we are on and it starts with **/**
+
+#### Navigation
+
+Now that we learned how to display the files on a directory (ls) let's learn how to move to another directory
+
+```
+danielvilla@Daniels-MacBook-Air:~$ cd docker
+danielvilla@Daniels-MacBook-Air:~/docker$ 
+```
+
+By the way, the **~** before the **$** sign indicates the home directory, all users have their own home directory, and is VERY IMPORTANT that you know that **/** indicates the root directory, off all the filesystem
+
+but now that we are on the docker directory how do we move back to the home directory, there are two ways:
+
+`danielvilla@Daniels-MacBook-Air:~/docker$ cd ..`
+
+and
+
+`danielvilla@Daniels-MacBook-Air:~/docker$ cd`
+
+**cd** without any arguments changes to your home directory no matter where you are
+
+You can also take advantage of the **~** symbol and use it to change to a directory inside your home 
+
+```
+danielvilla@Daniels-MacBook-Air:~/projects/selenium-projects/likeNeitor$ cd ~/docker
+danielvilla@Daniels-MacBook-Air:~/docker$ 
+```
+or use the absoluth path (in case the directory is not in your home)
+
+```
+danielvilla@Daniels-MacBook-Air:~$ cd /var/log/ppp/
+danielvilla@Daniels-MacBook-Air:/var/log/ppp$ 
+```
+
+But now let's say we are working on two very far directories
+`~/projects/react_app/config/` and `/etc/ssh/`
+
+#### More on ls command
+
+We already learn about **ls** but as we saw **ls** only showed us the list of file with no metadata
+
+```
+danielvilla@Daniels-MacBook-Air:~/docker/linux_tweet_app$ ls
+Dockerfile          README.md           index-new.html      index-original.html index.html          linux.png
+```
+
+but as a curious user that I am, I want to know the date the file was created, the size, the owner. To display it lets use the **-l** flag. Yes, we can use flags on commands to make them do more
```

Comment:
but as a curious user that I am, I want to know the date the file was created, the size `and` the owner. To display it lets use the **-l** flag. Yes, we can use flags on commands to make them do more`:`